### PR TITLE
RTA-286 Referrer

### PR DESCRIPTION
**Pull request by melgenek**
https://liveintent.atlassian.net/browse/RTA-286

### CODE REVIEW

**Review by jankoulaga — APPROVED** [review]

File: src/utils/page.js [hunk 1]
```diff
@@ -1,29 +1,46 @@
...
-  return (document.domain || (document.location && document.location.host)) || (window && window.location && window.location.host) || 'localhost'
+export function getPage (win = window) {
+  const ancestorOrigins = _safeGet(() => win.location.ancestorOrigins) || {}
+
+  const windows = []
+  let currentWindow
+  do {
+    currentWindow = currentWindow ? currentWindow.parent : win
+    windows.push(currentWindow)
+  } while (currentWindow !== top)
```
Comment (gipeshka) [inline src/utils/page.js:26]: Might be a bit shorter, if you prefer `for`
```suggestion
  const windows = [win]
  
  for (let currentWindow = win; currentWindow !== top; currentWindow = currentWindow.parent) {
    windows.push(currentWindow)
  }
```

**Review by gipeshka — APPROVED** [review]

File: src/utils/page.js (hunk 1 above)
Comment (melgenek) [reply src/utils/page.js:26]: The semantic is different. For loop doesn't keep the window #0, if it's already "top". My code does.

Comment (melgenek) [reply src/utils/page.js]: `const windows = [win]` I didn't manage to come up with this and my for loop didn't work :facepalm: 

:+1:

**Comment by jankoulaga** [comment]
@melgenek your linting is off ;)

File: src/utils/page.js (hunk 1 above)
Comment (melgenek) [reply src/utils/page.js]: this code is wrong

**Pull request merged**